### PR TITLE
s0, gpu fbank and resample

### PR DESCRIPTION
**Pull request by qmpzzpmq**

### CODE REVIEW

File: examples/librispeech/s0/conf/train_conformer.yaml [hunk 1]
```diff
@@ -47,6 +47,8 @@ collate_conf:
     speed_perturb: false
     feature_extraction_conf:
         feature_type: 'fbank'
+        from_gpu: true
+        sample_rate: 16000
```
Comment (robin1001) [inline examples/librispeech/s0/conf/train_conformer.yaml:51]: add a resample bool flag?
Maybe we will mix the 8k and 16k training by resampling or just ignoring the sample rate in the future.

**Review by robin1001 — CHANGES_REQUESTED** [review]
great job, please see inline.

File: examples/librispeech/s0/conf/train_conformer.yaml (hunk 1 above)
Comment (robin1001) [reply examples/librispeech/s0/conf/train_conformer.yaml:51]: and any results of the speedup by using GPU for fbank computation?

**Comment by qmpzzpmq** [comment]
still some work could be do later
1 fbank gpu batch computing
2. torch spec augment

**Comment by qmpzzpmq** [comment]
gpu fbank: 13min/100step
cpu fbank: 29min/100step

gpu: 8*V100
cpu: Intel(R) Xeon(R) Gold 6130 CPU

**Comment by whiteshirt0429** [comment]
> gpu fbank: 13min/100step
> cpu fbank: 29min/100step
> 
> gpu: 8*V100
> cpu: Intel(R) Xeon(R) Gold 6130 CPU
> dataset: personal dataset

perhaps results with dropping cache will be more reliable

**Comment by qmpzzpmq** [comment]
cpu fbank: 29min/100step： 88min wich cache drop

File: wenet/dataset/dataset.py [hunk 2]
```diff
@@ -182,38 +182,57 @@ def _extract_feature(batch, speed_perturb, wav_distortion_conf,
...
+            else:
+                waveform, org_sample_rate = torchaudio.load_wav(x[1])
+                if org_sample_rate != sample_rate:
+                    waveform = kaldi.resample_waveform(
+                        waveform, org_sample_rate, sample_rate)
+            if len(waveform.shape) > 1:
+                waveform = waveform.mean(dim=0)
             else:
-                waveform, sample_rate = torchaudio.load_wav(x[1])
+                waveform = waveform.squeeze(0)
```
Comment (placebokkk) [inline wenet/dataset/dataset.py:195]: why do squeeze here?

**Review by placebokkk — COMMENTED** [review]
Thanks for this great contribution! We will do an internal benchmark to verify this improvement and merge this PR.

**Pull request closed without merging**